v0.5/fastn-p1/src/parser/scanner.rs

```diff
@@ -134,18 +134,30 @@ impl Scanner {
 
     // returns the span from current position to the end of token
     pub fn token(&mut self, t: &'static str) -> Option<fastn_p1::Span> {
-        let mut count = 0;
-        for char in t.chars() {
+        // Get the length of the token to match
+        // t.chars().count(): Counts characters (Unicode scalar values).
+        let token_len = t.chars().count();
```

Comment:
Since `t` is assumed ascii, we can assume `token_len` and `t.len()` are same. Or we can stop assuming that `t` is ascii. Treating `token_len` and `t.len()` as possibly different at the same time when we assume t is ascii is wrong. 

We currently do not have any non ascii tokens, so while I feel like saying lets relax ascii requirement, our scanner can handle emoji etc, yay, but since we will never be using it, its better to keep the contract. 

Can we achieve the effect you are trying to achieve by simply fixing this line: `self.s_index = self.index` as `self.s_index += count;`?